bentoml/deployment/sagemaker.py

```diff
@@ -0,0 +1,181 @@
+    except ecr_client.exceptions.RepositoryNotFoundException:
+        ecr_client.create_repository(repositoryName=image_name)
+
+    print('Building docker image: {}'.format(image_name))
+    built_image = docker_client.images.build(path=snapshot_path, tag=image_name)
+    built_image[0].tag(ecr_tag)
+    print('Pushing image to AWS ECR at {}'.format(ecr_tag))
+    docker_client.api.push(repository=ecr_tag)
+    print('Finished pushing image: {}'.format(ecr_tag))
+    return ecr_tag
+
+
+def deploy_with_sagemaker(archive_path, additional_info):
+    """Deploy BentoML service to AWS Sagemaker.
+    Your AWS credential must have the correct permissions for sagemaker and ECR
+
+    :param archive_path: Path
+    :param additional_info: Dict
+    :return:
+    """
+
+    # 1. Create docker image and push to ECR
+    # 2. Create sagemaker model base on the ECR image
+    # 2. Create sagemaker endpoint configuration base on sagemaker model
+    # 3. Create sagemaker endpoint base on sagemaker endpoint configuration
+    bento_service = load(archive_path)
+    snapshot_path = generate_bentoml_deployment_snapshot_path(bento_service.name, 'aws-sagemaker')
+    shutil.copytree(archive_path, snapshot_path)
+    region = additional_info.get('region', DEFAULT_REGION)
+    execution_role_arn = get_arn_role_from_current_user()
+    ecr_image_path = create_push_image_to_ecr(bento_service, snapshot_path)
+    sagemaker_model_name = generate_aws_compatible_string('bentoml-' + bento_service.name + '-' +
+                                                          bento_service.version)
+    sagemaker_client = boto3.client('sagemaker', region_name=region)
+
+    sagemaker_model_info = {
+        "ModelName": sagemaker_model_name,
+        "PrimaryContainer": {
+            "ContainerHostname": sagemaker_model_name,
+            "Image": ecr_image_path,
+        },
+        "ExecutionRoleArn": execution_role_arn,
+    }
+    print('Creating sagemaker model {}'.format(sagemaker_model_name))
+    model_response = sagemaker_client.create_model(**sagemaker_model_info)
+    logger.info(model_response)
+
+    endpoint_config_name = generate_aws_compatible_string(bento_service.name + '-' +
+                                                          bento_service.version + '-configuration')
+    production_variants = [{
+        "VariantName": bento_service.name,
+        "ModelName": sagemaker_model_name,
+        "InitialInstanceCount": DEFAULT_INITIAL_INSTANCE_COUNT,
+        "InstanceType": DEFAULT_INSTANCE_TYPE,
+    }]
+    print('Creating sagemaker endpoint {} configuration'.format(endpoint_config_name))
+    create_endpoint_config_response = sagemaker_client.create_endpoint_config(
+        EndpointConfigName=endpoint_config_name, ProductionVariants=production_variants)
+    logger.info(create_endpoint_config_response)
+    print('Creating sagemaker endpoint {}'.format(bento_service.name))
```

Comment:
as discussed offline - make this logger.info as well

Reply:
done